enterprise/api_client/discovery.py

```diff
@@ -377,13 +383,16 @@ def _load_data(self, resource, default=DEFAULT_VALUE_SAFEGUARD, **kwargs):
 
         """
         default_val = default if default != self.DEFAULT_VALUE_SAFEGUARD else {}
-        result = get_edx_api_data(
-            api_config=CatalogIntegration.current(),
-            resource=resource,
-            api=self.client,
-            **kwargs
-        )
-        return result or default_val
+        try:
+            return get_edx_api_data(
+                api_config=CatalogIntegration.current(),
+                resource=resource,
+                api=self.client,
+                **kwargs
+            ) or default_val
+        except HttpClientError as exc:
```

Comment:
Does this need to be the `(ConnectionError, SlumberBaseException, Timeout)` trifecta like you have in the [ecommerce](https://github.com/edx/ecommerce/pull/1468/files#diff-0aa9e695700bf5edb4df6ef6ac088d71R87) PR?

Reply:
I'm honestly not sure about it. @haikuginger has more knowledge about Slumber -- he would know.

Reply:
@haikuginger What do you think? Seems like we should at least change this to `SlumberBaseException`.

Reply:
@douglashall, as far as I can tell, Slumber doesn't catch those other two, so let's include them.

Reply:
:+1: